### PR TITLE
7 post/api/articles/ id/comments

### PR DESCRIPTION
**Pull request by Spyabo**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -186,3 +186,50 @@ describe("GET: /api/articles/:article_id/comments", () => {
...
+
+  it("400: respond with an error for posting on an ID that is not possible", () => {
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid ID");
+      });
+  });
```
Comment (NRMSMITH) [inline __tests__/app.test.js:234]: A couple more tests for this one! 

201 - ignores unnecessary properties in a sent comment
400 - missing body or username
404 - username does not exist

File: controllers/comments.controller.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+exports.postComment = async (req, resp, next) => {
+  const article_id = parseInt(req.params.article_id);
+  const comment = {
+    username: req.body.username,
+    body: req.body.body,
+    article_id,
+  };
+  insertComment(comment)
+    .then(async (data) => {
+      resp.status(201).send(data);
```
Comment (NRMSMITH) [inline controllers/comments.controller.js:12]: try to send this on a key of 'postedComment' or something like that, honestly for readability more than anything :)

File: models/comments.model.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const db = require("../db/connection.js");
+const { checkArticleExists } = require("../models/fetchArticles.model.js");
+
+exports.insertComment = async (comment) => {
+  const { username, body, article_id } = comment;
+  if (await checkArticleExists(article_id)) {
```
Comment (NRMSMITH) [inline models/comments.model.js:6]: I'm not sure if this is needed here - as it will be a psql error if the article_id does not exist?

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
A couple of edits to make - mostly testing but one or two in the model as well :)

File: models/users.model.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const db = require("../db/connection.js");
+
+async function checkUserExists(username) {
```
Comment (itsaliabdallah) [inline models/users.model.js:3]: This function wouldn't be needed.
If a you send a comment with a username which isn't in the database PSQL will throw an error.

**Review by itsaliabdallah — CHANGES_REQUESTED** [review]
Testing all looks good now!

As previous comments suggest, you do not need to check if the Article ID exists before making your db.query as PSQL will return an Error, same for the username.

**Pull request merged**